### PR TITLE
Fix methods spec harness

### LINKED ISSUE #29: Fix Methods - Spec Harness

### PR DESCRIPTION
**Pull request by westonio**
## Describe your changes in detail
- Add require_relative for all lib files to work with the spec harness
- Correct the following methods and relevant tests to pass spec harness:
> `percentage_home_wins`, `percentage_visitor_wins`, `percentage_ties`
> `avg_goals_by_team`, `best_offense`, 'worst_offense

## Checklist before requesting a review
- [ x ] I have performed a self-review of my code
- [ x ] I have added thorough tests
- [ x ] I AM MERGING WITH Westonio/futbol

## Anything else noteworthy? (E.g. note anything that might need refactoring or special attention later)
worst/best offense methods have been refactored!

### CODE REVIEW

File: lib/raw_stats.rb [hunk 1]
```diff
@@ -12,26 +15,4 @@ def initialize(data)
...
-
-  # def parse_teams(data) = CSV.foreach(data, headers:true, header_converters: :symbol) do |line|
-  #   @teams <<
-  # end
-
-  # def parse_game_teams(data) = CSV.foreach(data, headers:true, header_converters: :symbol) do |line|
-
-  # end
\ No newline at end of file
+end
```
Comment (pcbennett108) [inline lib/raw_stats.rb:18]: Thanks for taking out these comments. I also think they are no longer necessary;.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -93,24 +93,20 @@ def avg_goals_by_team
...
 
   def worst_offense
-    lowest = avg_goals_by_team.min_by do |team_id, avg_goals|
-      avg_goals
-    end
-    team = @teams.find { |team| team.team_id == lowest[0]}
-    team.team_name
+    lowest = avg_goals_by_team.min_by { |team_id, avg_goals|avg_goals }
+    worst = @teams.find { |team| team.team_id == lowest[0]}
+    worst.team_name
```
Comment (pcbennett108) [inline lib/stat_tracker.rb:109]: I like the variable renaming to better reflect the data.

File: lib/raw_stats.rb (hunk 1 above)
Comment (fadwil) [reply lib/raw_stats.rb:18]: Good call on cleaning that up, I think we are set on the direction we want to go in for raw stats

File: lib/stat_tracker.rb (hunk 2 above)
Comment (fadwil) [reply lib/stat_tracker.rb:109]: Good stuff! This looks clean.

**Review by fadwil — APPROVED** [review]

**Review by pcbennett108 — COMMENTED** [review]
Thanks for setting up and trying out the spec harness.
Good catch on adjusting for the differences in the data sets.

**Pull request merged**